api/src/middleware/__tests__/assessmentsRouter.ts

```diff
@@ -59,17 +81,520 @@ describe('assessmentsRouter', () => {
+        responses: [
+          {
+            id: 1,
+            answer_id: 1,
+            assessment_id: 1,
+            submission_id: 2,
+            question_id: 1,
+          },
+        ],
+      };
+      const response: SubmittedAssessment = {
+        curriculum_assessment: curriculumAssessment,
+        program_assessment: programAssessment,
+        submission: assessmentSubmission,
+      };
+
+      mockPrincipalId(participantPrincipalId);
+      appAgent
+        .get(`/${programAssessmentId}/submissions/${submissionId}`)
+
+        .expect(200, itemEnvelope(response), err => {
+          expect(mockProgramAssessmentById).toHaveBeenCalledWith(
+            programAssessmentId
+          );
+          expect(mockGetCurriculumAssessmentById).toHaveBeenCalledWith(
+            programAssessmentId,
+            true,
+            true
+          );
+
+          expect(mockFindRoleInProgram).toHaveBeenCalledWith(
+            facilitatorPrincipalId,
+            programAssessment.program_id
+          );
+          expect(mockSubmissionDetails).toHaveBeenCalledWith(
+            programAssessmentId,
+            submissionId,
+            true
+          );
+          done(err);
+        });
+    });
+
+    it('should respond with a bad request error if given an invalid assessment id', done => {
+      const programAssessmentId = 'test',
+        submissionId = 1;
+
+      appAgent
+        .get(`/${programAssessmentId}/submissions/${submissionId}`)
+        .expect(400, done);
+    });
+
+    it('should respond with a NotFoundError if the assessment ID was not found in the database', done => {
+      const programAssessmentId = 7,
+        submissionId = 1;
+
+      mockGetProgramIdByProgramAssessmentId.mockResolvedValue([]);
+      appAgent
+        .get(`/${programAssessmentId}/submissions/${submissionId}`)
+        .expect(
+          404,
+          errorEnvelope('The requested resource does not exist.'),
+          err => {
+            expect(mockGetProgramIdByProgramAssessmentId).toHaveBeenCalledWith(
+              programAssessmentId
+            );
+            done(err);
+          }
+        );
+    });
+
+    it('should respond with a bad request error if given an invalid submission id ', done => {
+      const programAssessmentId = 1,
+        submissionId = 'test';
+
+      appAgent
+        .get(`/${programAssessmentId}/submissions/${submissionId}`)
+        .expect(400, done);
+    });
+
+    it('should respond with a NotFoundError if the submission ID was not found in the database ', done => {
+      const programAssessmentId = 1,
+        submissionId = 8;
+
+      mockSubmissionDetails.mockResolvedValue([]);
+      appAgent
+        .get(`/${programAssessmentId}/submissions/${submissionId}`)
+        .expect(
+          404,
+          errorEnvelope('The requested resource does not exist.'),
+          err => {
+            expect(mockSubmissionDetails).toHaveBeenCalledWith(
```

Comment:
This should have the same mock function calls and mock function call responses as your "should show a participant their submission information for an in-progress assessment without including the correct answers" test, with only the mockSubmissionDetails function returning a different value of an empty array.